pkgs/applications/science/math/sage/sage-src.nix

```diff
@@ -70,11 +70,16 @@ stdenv.mkDerivation rec {
   # Patches needed because of particularities of nix or the way this is packaged.
   # The goal is to upstream all of them and get rid of this list.
   nixPatches = [
-    # Make sure py2/py3 tests are only run when their expected context (all "sage"
-    # tests) are also run. That is necessary to test dochtml individually. See
-    # https://trac.sagemath.org/ticket/26110 for an upstream discussion.
-    # TODO: Determine if it is still necessary.
-    ./patches/Only-test-py2-py3-optional-tests-when-all-of-sage-is.patch
+    # Since https://trac.sagemath.org/ticket/32174, some external features are
+    # marked as "safe" and get auto-detected, in which case the corresponding
+    # optional tests are executed. We disable auto-detection of safe features if
+    # we are doctesting with an "--optional" argument which does not include
```

Comment:
This continues the tradition of the previous patch, but there are only three files that contain `dochtml` (now called `sagemath_doc_html`) tests, namely `src/sage/features/sagemath.py`, `src/sage/docs/conf.py` and `src/sage/misc/sagedoc.py`. I even had to cheat to make the first of those three files pass (see below).